### PR TITLE
Add watermark for release archive

### PR DESCRIPTION
**Pull request by hivepress**

### CODE REVIEW

File: includes/controllers/class-admin.php [hunk 1]
```diff
@@ -57,6 +57,14 @@ public function __construct( $args = [] ) {
 						'action' => [ $this, 'deactivate_plugin' ],
 						'rest'   => true,
 					],
+
+					'plugin_watermark_action'    => [
+						'base'   => 'admin_base',
+						'path'   => '/watermark-plugin',
+						'method' => 'POST',
```
Comment (hivepress) [inline includes/controllers/class-admin.php:64]: Please change this to a simple GET route, a non-REST one (so you can test this simply by URL).

File: includes/controllers/class-admin.php [hunk 2]
```diff
@@ -65,6 +73,34 @@ public function __construct( $args = [] ) {
...
 	}
 
+	public function watermark_plugin( $request ) {
+		$license_key = 'test';
+		$file_path   = '/test/test.zip';
+
+		if ( file_exists( $file_path ) ) {
+
+			// Open the existing ZIP file
+			$zip = new \ZipArchive();
```
Comment (hivepress) [inline includes/controllers/class-admin.php:83]: Is this a built-in PHP library?

File: includes/controllers/class-admin.php [hunk 3]
```diff
@@ -65,6 +73,34 @@ public function __construct( $args = [] ) {
...
+
+			if ( $zip->open( $file_path ) === true ) {
+				$zip->setArchiveComment( $license_key );
```
Comment (hivepress) [inline includes/controllers/class-admin.php:86]: Is it easy to detect and remove for pirates, e.g. via the 7Zip or other software UI?

Comment (hivepress) [reply includes/controllers/class-admin.php:86]: The watermark should be as hidden as possible blended into the file or if not then embedded into the obfuscated code of one of the inner archive files.

File: includes/controllers/class-admin.php (hunk 2 above)
Comment (kuznev) [reply includes/controllers/class-admin.php:83]: Yes

**Pull request closed without merging**